packages/babel-traverse/src/path/comments.js

```diff
@@ -5,6 +5,9 @@
  */
 
 export function shareCommentsWithSiblings() {
+  // NOTE: this assumes numbered keys
+  if (typeof this.key === "string") return;
```

Comment:
I wonder what value does the `key` get in a `LabeledStatement`... or does the sharing happen from the `ForOfStatement` side? That would be bad 😆